packages/lesswrong/server/cacheControlMiddleware.ts

```diff
@@ -0,0 +1,70 @@
+import type { AddMiddlewareType } from './apolloServer';
+import { parsePath, parseRoute } from '../lib/vulcan-core/appContext';
+import { getUserFromReq } from './vulcan-lib/apollo-server/context';
+import express from 'express';
+import { getCookieFromReq } from './utils/httpUtil';
+import { DatabaseServerSetting } from './databaseSettings';
+import { RouterLocation } from './vulcan-lib';
+import { Posts } from '../lib/collections/posts';
+
+const swrCachingEnabledSetting = new DatabaseServerSetting<boolean>('swrCaching.enabled', false)
+
+/**
+ * Whether stale-while-revalidate caching is enabled on this specific route. To be removed once
+ * the kinks have been worked out
+ */
+const experimentalCachingEnabled = async (parsedRoute: RouterLocation): Promise<boolean> => {
+  if (parsedRoute.currentRoute?.name !== "posts.single") return false;
+  const postId = parsedRoute.params?._id
+  if (!postId) return false;
+
+  const post = await Posts.findOne({_id: postId}, {}, {swrCachingEnabled: 1})
```

Comment:
I'm not sure that would be faster here, because it will only load one post per request, so it won't be able to do any batching. I guess it could result in caching the post for later in the request, but we only need one field here so I'm not sure if fetching the full post object here would be faster either. Anyway I'm going to leave it as is because this will hopefully be removed within a week anyway